node/pkg/db/rediscriber_test.go

```diff
@@ -0,0 +1,103 @@
+package db
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/redis/go-redis/v9"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewRediscriber(t *testing.T) {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	host := os.Getenv("REDIS_HOST")
+	port := os.Getenv("REDIS_PORT")
+
+	done := make(chan struct{})
+	rediscriber, err := NewRediscriber(ctx,
+		WithRedisHost(host),
+		WithRedisPort(port),
+		WithRedisChannels([]string{"test-channel"}),
+		WithRedisRouter(func(msg *redis.Message) error {
+			assert.Equal(t, "test-channel", msg.Channel)
+			assert.Equal(t, "test-message", msg.Payload)
+			close(done)
+			return nil
+		}),
+	)
+	require.NoError(t, err)
+	defer rediscriber.client.Close()
+
+	// Start the Rediscriber
+	started := make(chan struct{})
+	go func() {
+		err := rediscriber.Start(ctx)
+		require.NoError(t, err)
+		close(started)
+	}()
+
+	<-started
+
+	// Publish a test message
+	err = rediscriber.client.Publish(ctx, "test-channel", "test-message").Err()
+	require.NoError(t, err)
+
+	select {
+	case <-done:
+		return
+	case <-time.After(5 * time.Second):
+		t.Fatal("Test timed out waiting for message processing")
+	}
+}
+
+func TestRediscriber_ErrorHandling(t *testing.T) {
```

Comment:
minor suggestion, how about removing the underscore to keep the naming consistent?